### ISSUE TITLE
Apple TV is found twice - plugin fails

### ISSUE THREAD

**Issue by normen** [lprhodes/homebridge-apple-tv#14]
After I first couldn't discover my AppleTV at all (see #9) it seems like the plugin is discovering my AppleTV twice now :)

This causes the plugin to crash homebridge because it tries to create two pair switches with the same name:

```
[2018-4-6 18:11:35] [Apple TV] [INFO] Discovered Apple TV (Apple TV) at 192.168.2.113
[2018-4-6 18:11:35] [Apple TV] [INFO] Added pair switch for Apple TV (Apple TV) at 192.168.2.113
[2018-4-6 18:11:35] [Apple TV] [INFO] Discovered Apple TV (Apple TV) at 192.168.2.113
[2018-4-6 18:11:35] [Apple TV] [INFO] Added pair switch for Apple TV (Apple TV) at 192.168.2.113
[2018-4-6 18:11:35] [Apple TV] [INFO] Connecting to "Apple TV"
[2018-4-6 18:11:35] [cmdSwitch2] 3D Drucker is off.
[2018-4-6 18:11:36] [Apple TV] [INFO] Connected to "Apple TV"
[2018-4-6 18:11:36] [Apple TV] Initializing platform accessory 'Pair Apple TV'...
[2018-4-6 18:11:36] [Apple TV] Initializing platform accessory 'Pair Apple TV'...
[2018-4-6 18:11:36] [Apple TV] 
```

causing this:

```
(node:8016) UnhandledPromiseRejectionWarning: Unhandled promise rejection (rejection id: 1): Error: Cannot add a bridged Accessory with the same UUID as another bridged Accessory: f958068f-b583-49f2-bd3b-5f17f6e254b6
(node:8016) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.
```

This happens with a clean homebridge with only the Apple TV plugin enabled.

Edit: When I get the credentials manually using the command line tool and set the setEnablePairSwitches to false the plugin works as expected.

**Comment by EtienneMD** [comment]
How exactly to you get the credentials manually? I am facing the same issue, and would like to use the plugin until this gets resolved. 
Thanks.

**Comment by normen** [comment]
Using the appletv command line tool, check the referenced issue.

**Comment by crzdg** [comment]
Sorry, where do you set the option to disable automatic discovery?
config.json of Homebridge? Should be in the AppleTV platform section right?

EDIT
Sorry, it was to early yesterday.
Pair devices manually and add them to the config in advance. Also set the "showPairSwitches" option to false after platform and name in the platform section. The auto discovery will still run, but it will not try to add the switches so it will not crash because of two same named switches.

**Comment by NorthernMan54** [comment]
I had a similar problem and resolved it by adding "showPairSwitches": false to my config.json

**Comment by MsCopernic** [comment]
Hi @normen: "Using the appletv command line tool, check the referenced issue.". Can you please add a link? Thanks in advance ..

**Comment by normen** [comment]
Hi @MsCopernic, can you look at the first post for the link? Thanks in advance...

**Comment by MsCopernic** [comment]
Hi @normen, thanks, after installing the command tool, I'm getting the following error: 

iMac-de-Jorge:~ jorgelperez$ appletv pair
? Which Apple TV would you like to pair with? Alcoba (192.168.1.57:49154)
✔ Connecting to Alcoba
✔ Initiating Pairing
? Enter the 4-digit pin that's currently being displayed on Alcoba 0864
✖ Completing Pairing
Cannot read property 'toString' of undefined
iMac-de-Jorge:~ jorgelperez$ 

Any idea?

Regards

**Comment by normen** [comment]
Hi @MsCopernic, it looks like it can't read the property "toString" of a nullpointer.

**Comment by MsCopernic** [comment]
Yup, that's clear for me @normen, but from where and why? How do I fix this problem?

**Comment by MsCopernic** [comment]
Hi @normen, I tried again today and it worked ??? I have made no changes. Thanks anyway.

**Comment by MsCopernic** [comment]
As said before or some where else, the AppleTV responds so fast, that they is/are detected twice:

❯ Alcoba (192.168.1.57:49154) 
  Estudio Merce (192.168.1.53:49154) 
  Alcoba (192.168.1.57:49154) 
  Estudio Merce (192.168.1.53:49154)

The only difference from what I did yesterday, is that this time I selected the ones from the second detection. I assume that if you choose the first ones, the information from that detection is gone, that may conduce to the null pointer error.

Regards

**Comment by Tangston311** [comment]
I also encountered this issue on my Pi and resolved it via the suggestions above.  I found the manual pairing process to be a bit confusing at first, so if it helps anyone:

First I found the file location of the plugin on my Pi.  For me it was: /usr/lib/node_modules/homebridge-apple-tv/node_modules/node-appletv/bin

Then I opened up a terminal window _from that folder location_ and ran: `cd node_modules/node-appletv/bin`

Then I ran: `./appletv pair`

My AppleTV popped up twice - I chose to pair with the second one and followed the on-screen prompts.  

Lastly, I added `"showPairSwitches": false` to my config file directly under the "platforms" section and pasted the pairing code I got above into the appropriate credentials place.

**Comment by Crox77** [comment]
So I'm getting this error and Homebridge not starting I have followed the above posts and paired my ATV manually but HB still will not start. Here is the error message and the part of my config file:

[3/13/2019, 9:23:25 PM] [Apple TV] [INFO] Discovered Apple TV (Crox Apple TV 4K) at 192.168.0.25
[3/13/2019, 9:23:25 PM] [Apple TV] [INFO] Discovered Apple TV (Crox Apple TV 4K) at 192.168.0.25
[3/13/2019, 9:23:25 PM] [Apple TV] [INFO] Connecting to "Apple TV 4K"
[3/13/2019, 9:23:25 PM] [Apple TV] [INFO] Connected to "Apple TV 4K"
(node:4142) UnhandledPromiseRejectionWarning: TypeError: Cannot read property 'split' of undefined
    at connectToDevices (/usr/local/lib/node_modules/homebridge-apple-tv/helpers/devices.js:31:29)

(node:4142) UnhandledPromiseRejectionWarning: Unhandled promise rejection. This error originated either by throwing inside of an async function without a catch block, or by rejecting a promise which was not handled with .catch(). (rejection id: 1)
(node:4142) [DEP0018] DeprecationWarning: Unhandled promise rejections are deprecated. In the future, promise rejections that are not handled will terminate the Node.js process with a non-zero exit code.


{
			"platform": "AppleTV",
			"showPairSwitches": false,
			"name": "Apple TV",
			"hideWelcomeMessage": true,
			"devices": [{
					"id": "lounge",
					"name": "Apple TV 4K",
					"credentials": "2ECA59F4-7BE6-441B-9F82-8CD602C5A657:30323445354434432d384630412d343342442d423044432d354235364535334635303236:33336133643661352d383663392d343961322d393365322d316139316438346666383064:0a885827cf204e1ae151a4530cf4b91fad22ee0b9c3b0290823fb6e3839ae72f:6295837a2dd0d0d34cc05af55bf69e41e9f64632c1d9e691b69a47db291da41c"
				},
Any more ideas please?

**Comment by Crox77** [comment]
Fixed it, changed the name of Apple TV and now working

**Comment by chrismaurer** [comment]
Which name did you change @Crox77 ? The name under “devices”:[] or the name under “platform”: ?

Thanks!
